app/helpers/address_helper.rb

```diff
@@ -88,17 +87,38 @@ def make_address_hash(address)
 
   def update_support_address(selected, given)
     given[:uprn] = selected[:uprn] if compare(selected, given)
-    given.delete(:sao_present)
     { support_address: given }
   end
 
-  def compare(selected, given)
-    return false if selected.empty?
+  def sanitize_postcode(postcode)
+    postcode.gsub(/[^0-9a-zA-Z]+/, "")
+  end
+
+  def sanitize_address(address)
+    return [] if address.blank?
+
+    if address.values.any?
+      copy_of_address = address
+      copy_of_address[:postcode] = sanitize_postcode(address[:postcode])
+
+      copy_of_address
+        .with_indifferent_access
+        .except(:uprn, :county, :town_city)
+        .values
+        .map(&:upcase).join(" ")
+        .gsub(/[^0-9A-Z]+/, " ")
+        .split(" ")
+        .sort
+        .compact
+    end
+  end
 
-    line_one_and_two = address_comparison_line_builder(selected, LINE_1_2_AND_POSTCODE) == address_comparison_line_builder(given, LINE_1_2_AND_POSTCODE)
+  def compare(selected, given)
+    return false if selected.blank?
```

Comment:
The expectation is that we would get false.  The reason for not testing it explicitly is because the compare will handle that, and we're really only tested that the user has selected an address at this point.

Reply:
Good point - I've added explicit tests for either the edited or ordnance addresses being blank.